Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -1110,4 +1110,61 @@ theorem isBaseChange : IsBaseChange A f := by
 
 end IsLocalizedModule
 
+section Basis
+
+open Submodule
+
+variable (Rₛ : Type*) [CommRing Rₛ] [Algebra R Rₛ] [IsLocalization S Rₛ] (Mₛ : Type*)
+  [AddCommGroup Mₛ] [Module R Mₛ] {M : Submodule R Mₛ} [IsLocalizedModule S M.subtype]
```

Comment:
Is there an obstruction generalizing `M.subtype` to an arbitrary linear map (which isn't necessarily injective)?

Reply:
Yes, there is an obstruction: `LinearIndependent.map' ` in `Basis.isLocalizedModule` needs an injective map.

Reply:
I think we should prove LinearIndependent assuming LinearIndependent only, and it shouldn't be too hard to adapt the proof of [LinearIndependent.localization](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Localization/Module.html#LinearIndependent.localization) which should work in the non-injective case.

Reply:
Indeed, that may work. I'll be a bit busy in the next few days but I'll see what I can prove after that.